src/transformers/models/speecht5/tokenization_speecht5.py

```diff
@@ -142,6 +148,8 @@ def __setstate__(self, d):
 
     def _tokenize(self, text: str) -> List[str]:
         """Take as input a string and return a list of strings (tokens) for words/sub-words"""
+        if self.normalize:
+            text = self._normalize(text)
```

Comment:
This should go in the `prepare_inputs_for_tokenization`. This way it could support normalizing on the fly with kwargs, and only normalize once if there are special tokens (instead of normalizing each substrings)

Reply:
Makes sense, will do that!

Reply:
@ArthurZucker Should I make a new function called `prepare_inputs_for_tokenization`?

Reply:
It's actually `prepare_for_tokenization` which should be overloaded!

Reply:
Gotcha. This is what I am doing currently @ArthurZucker;

```python
# ... rest of the code in Tokenizer class
    @property
    def normalizer(self):
        return self.normalizer

    @normalizer.setter
    def normalizer(self):
        self.normalizer = EnglishNumberNormalizer()
# ... other code

    def prepare_for_tokenization(self, text, is_split_into_words=False, **kwargs):
        normalize = kwargs.pop("normalize", self.normalize)
        if is_split_into_words:
            text = " " + text
        if normalize:
            text = self.normalizer(text)
        return (text, kwargs)
```

Am I on the right track / any suggestions?